sentrysql/stmt.go

```diff
@@ -0,0 +1,159 @@
+package sentrysql
+
+import (
+	"context"
+	"database/sql/driver"
+	"errors"
+
+	"github.com/getsentry/sentry-go"
+)
+
+type sentryStmt struct {
+	originalStmt driver.Stmt
+	query        string
+	ctx          context.Context
+	config       *sentrySQLConfig
+}
+
+// Make sure sentryStmt implements driver.Stmt interface.
+var _ driver.Stmt = (*sentryStmt)(nil)
+
+func (s *sentryStmt) Close() error {
+	return s.originalStmt.Close()
+}
+
+func (s *sentryStmt) NumInput() int {
+	return s.originalStmt.NumInput()
+}
+
+//nolint:dupl
+func (s *sentryStmt) Exec(args []driver.Value) (driver.Result, error) {
+	parentSpan := sentry.SpanFromContext(s.ctx)
+	if parentSpan == nil {
+		return s.originalStmt.Exec(args) //nolint:staticcheck // We must support legacy clients
+	}
+
+	span := parentSpan.StartChild("db.sql.exec", sentry.WithDescription(s.query))
+	s.config.SetData(span, s.query)
+	defer span.Finish()
+
+	result, err := s.originalStmt.Exec(args) //nolint:staticcheck // We must support legacy clients
+	if err != nil {
+		span.Status = sentry.SpanStatusInternalError
+		return nil, err
+	}
+
+	span.Status = sentry.SpanStatusOK
+
+	return result, nil
```

Comment:
This section is not covered by tests, yet I don't know how to make it so.

I've added this check beforehand:
```go
if s.ctx == nil && ctx != nil {
  s.ctx = ctx
}
```

But it doesn't seem to help anyway.